technic/machines/compat/tools.lua

```diff
@@ -0,0 +1,35 @@
+
+-- This wont give full compatibility but allows using old style technic.register_power_tool tool registration function.
+-- Tools still have to read charge value using technic.get_RE_charge, allows easier compatibility with official Technic
+-- mod for some tools: a lot less changes required for compatibility but tool will keep some unnecessary meatadata.
+--
+-- To make tool fully compatible replace minetest.register_tool with technic.register_power_tool and add `max_charge`
+-- field for tool definition. Fields `wear_represents` and `on_refill` can also be removed if using defaults.
+--
+local register_power_tool = technic.register_power_tool
+function technic.register_power_tool(itemname, itemdef)
+	if type(itemdef) == "number" then
+		minetest.log("warning", "Deprecated technic.register_power_tool use. Setting max_charge for "..itemname)
+		technic.power_tools[itemname] = itemdef
+		minetest.register_on_mods_loaded(function()
+			minetest.log("warning", "Deprecated technic.register_power_tool use. Ensuring fields for "..itemname)
+			local redef = minetest.registered_items[itemname]
+			if redef and redef.wear_represents == "technic_RE_charge" and not redef.on_refill then
+				-- Override power tools that called register_power_tool but do not have on_refill function defined
+				minetest.override_item(itemname, {
+					on_refill = function(stack)
+						technic.set_RE_charge(stack, stack:get_definition().max_charge or 10000)
+						return stack
+					end,
+					max_charge = itemdef,
+				})
+				minetest.log("warning", "Updated on_refill and max_charge for "..itemname)
+			end
+		end)
+	else
+		return register_power_tool(itemname, itemdef)
+	end
+end
+
+-- Alias set set_RE_wear, many tools calls this to set wear value which is also handled by set_RE_charge
+technic.set_RE_wear = technic.set_RE_charge
```

Comment:
I think it could be good to add a deprecation warning here too.
```suggestion
function technic.set_RE_wear(stack, charge)
	minetest.log("warning", "Use of deprecated function technic.set_RE_wear. Use technic.set_RE_charge instead.")
	technic.set_RE_charge(stack, charge)
end
```

Reply:
Maybe include tool name too to make it obvious where it comes from. I'll smelt that into first commit.

Reply:
I did intentionally left "Use technic.set_RE_charge instead." part out. I think it is better to encourage using documentation (or even read code) because there technic.set_RE_charge might not be best option for upgrade. Documentation should of course be updated too.